client/src/components/Schedule/ListView.tsx

```diff
@@ -1,17 +1,219 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Collapse, Badge, Button, Row, Typography, Col, Tooltip } from 'antd';
+import { LeftOutlined, RightOutlined } from '@ant-design/icons';
+import css from 'styled-jsx/css';
+import moment from 'moment-timezone';
 import { CourseEvent } from 'services/course';
+import { dateWithTimeZoneRenderer, renderTag } from 'components/Table';
+import { EventTypeColor, EventTypeToName } from './model';
+
+const { Panel } = Collapse;
+const { Text } = Typography;
+
+const currentDayColor = '#ffd591';
+const previuosDaysColor = '#fff1f0';
+const nextDaysColor = '#d9f7be';
+const WEEK = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
+const weekLength = WEEK.length;
 
 type Props = {
   data: CourseEvent[];
   timeZone: string;
 };
 
-export function ListView({ data, timeZone }: Props) {
+export const ListView = ({ data, timeZone }: Props): React.ReactElement => {
+  const [currentWeek, setCurrentWeek] = useState(0);
+  const [startWeekDate, setStartWeekDate] = useState('');
+  const [endWeekDate, setEndWeekDate] = useState('');
```

Comment:
Looks like start and end date are used only in render. So there is no need to store them in state, let's calculate these data directly in render, but memoized version(using useMemo or useCallback for example) for optimizing performance.

Reply:
done